cassandra-diagnostics-connector21/src/main/java/io/smartcat/cassandra/diagnostics/connector/ConnectorProxyImpl.java

```diff
@@ -0,0 +1,328 @@
+package io.smartcat.cassandra.diagnostics.connector;
+
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.returns;
+import static net.bytebuddy.matcher.ElementMatchers.takesArguments;
+
+import java.lang.instrument.Instrumentation;
+import java.lang.reflect.Modifier;
+
+import org.apache.cassandra.cql3.CQLStatement;
+import org.apache.cassandra.cql3.QueryOptions;
+import org.apache.cassandra.cql3.statements.ParsedStatement;
+import org.apache.cassandra.service.QueryState;
+import org.apache.cassandra.transport.messages.ResultMessage;
+import org.apache.cassandra.utils.MD5Digest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.googlecode.concurrentlinkedhashmap.ConcurrentLinkedHashMap;
+
+import akka.actor.ActorRef;
+import akka.actor.ActorSystem;
+import io.smartcat.cassandra.diagnostics.actor.ActorFactory;
+import io.smartcat.cassandra.diagnostics.actor.messages.Notification;
+import io.smartcat.cassandra.diagnostics.config.Configuration;
+import net.bytebuddy.ByteBuddy;
+import net.bytebuddy.agent.builder.AgentBuilder;
+import net.bytebuddy.agent.builder.AgentBuilder.InitializationStrategy;
+import net.bytebuddy.agent.builder.AgentBuilder.RedefinitionStrategy;
+import net.bytebuddy.agent.builder.AgentBuilder.Transformer;
+import net.bytebuddy.agent.builder.AgentBuilder.TypeStrategy;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.NamedElement;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.dynamic.DynamicType.Builder;
+import net.bytebuddy.implementation.Implementation;
+import net.bytebuddy.matcher.ElementMatcher;
+import net.bytebuddy.matcher.ElementMatchers;
+
+/**
+ * Connector implementation.
+ */
+public class ConnectorProxyImpl implements ConnectorProxy {
```

Comment:
It might be personal, but I don't like implementation classes having *Impl sufix. This does not explain anything specific to implementation of concrete interface. If there is nothing specific for this implementation, it is better to call it DefaultConnectorProxy or BasicConnectorProxy